### PR TITLE
Fix inverse in controlled gates

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

`inverse()` in controlled gates do not keep `ctrl_state`.
I have fixed this and added test.

### Details and comments

This will solve an issue in Aqua.
https://github.com/Qiskit/qiskit-aqua/issues/1215

For example,
```python
from qiskit.circuit.library import CSwapGate

print(CSwapGate(ctrl_state=0) == CSwapGate(ctrl_state=0).inverse())
```
outputs `False`, but this should output `True`.

### CODE REVIEW

**Comment by ewinston** [comment]
I think it would be safer to not preserve the label under inverse, and to set it explicitly.

**Comment by ikkoham** [comment]
Thank you for the comments.

I see. Which do you think is better?
```python
    def inverse(self, label=None):
        """Return inverted CH gate (itself)."""
        label = label if label is not None else self.label
        return CHGate(label=label, ctrl_state=self.ctrl_state)  # self-inverse
```
or
```python
    def inverse(self, label=None):
        """Return inverted CH gate (itself)."""
        return CHGate(label=label, ctrl_state=self.ctrl_state)  # self-inverse
```

**Comment by 1ucian0** [comment]
It's my understanding that @ewinston  prefers the second option.

**Comment by ikkoham** [comment]
Sorry. Such a lint error has occurred. 
```
W0221: Parameters differ from overridden 'inverse' method (arguments-differ)
```
We have a two choises:
1. Do not add label to sub classes.
2. Add label to the super class (Instruscion Class)

2 needs big changes.

I am not familiar with `label` and I cannot judge exactly which is the preferred specification.

File: releasenotes/notes/fix-controlledgate-inverse-4f87ffd362127186.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fix bugs of `inverse` methods in controlled gates,
+    which do not preserve `ctrl_state`.
```
Comment (1ucian0) [inline releasenotes/notes/fix-controlledgate-inverse-4f87ffd362127186.yaml:5]:
```suggestion
    Fix bugs of the concrete implementations of meth:`~qiskit.circuit.ControlledGate.inverse` method which do not preserve the `ctrl_state` parameter.
```

**Pull request merged**